EmbeddedScripts.CSharp/CodeCompiler.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using EmbeddedScripts.Shared;
+
+namespace EmbeddedScripts.CSharp
+{
+    internal class CodeCompiler
+    {
+        public CodeCompiler(string code, Container container)
+        {
+            Container = container;
+            Compilation = Compile(code);
+        }
+
+        public CSharpCompilation Compilation { get; set; }
+
+        private CSharpCompilation Compile(string code)
+        {
+            var syntaxTree = CSharpSyntaxTree.ParseText(code);
+
+            var assemblyName = Path.GetRandomFileName();
+
+            var references = GenerateMetadataReferences();
+
+            var opts = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary);
+
+            return CSharpCompilation.Create(
+                assemblyName,
+                new[] { syntaxTree },
+                references,
+                opts
+            );
+        }
+
+        private List<MetadataReference> GenerateMetadataReferences()
+        {
+            var dotNetCoreDir = Path.GetDirectoryName(typeof(object).Assembly.Location);
+
+            return Container.VariableAliases
```

Comment:
Maybe over engineering?

Reply:
No, but hard to read. It would be better to extract 3 Select stmts to the private function `MetadataReference GetMetadataReference()`